languages/wikipron/src/write.py

```diff
@@ -0,0 +1,97 @@
+        # the name we are adding.
+        if (
+            length == 3
+            or wiki_name > wiki_name_of_last_entry
+        ):
+            # Write the input string to the end
+            source.write(row_string)
+        else:
+            for i in range(3, length):
+                isolated_name = readme_list[i].split("|")[4][1:-1]
+                # Replaces old row.
+                if (
+                    isolated_name == wiki_name
+                    and row_string.split("|")[6][1:-1] in readme_list[i]
+                ):
+                    readme_list[i] = row_string
+                    break
+                # Inserts new row.
+                if isolated_name > wiki_name:
+                    readme_list.insert(i, row_string)
+                    break
+            # Rewrite the entire README.
+            source.seek(0)
+            source.truncate()
+            source.write("".join(readme_list))
+
+
+def main():
+    with open(LANGUAGES_PATH, "r") as source:
+        languages = json.load(source)
+    readme_tsv_list = []
+    path = "../tsv"
+    for file in os.listdir(path):
+        # Filters out sole README.md file
+        if file.endswith(".tsv"):
+            with open(f"{path}/{file}", "r") as tsv:
+                num_of_entries = sum(1 for line in tsv)
+            iso639_code = file[: file.index("_")]
+            transcription_level = file[
+                file.rindex("_") + 1 : file.index(".")
+            ].capitalize()
+            wiki_name = languages[iso639_code]["wiktionary_name"]
+            # Assumes we will not remove eng and spa tsv files
+            # collected in previous big scrape (with no dialect specification)
+            if "dialect" in languages[iso639_code]:
+                # Check to make sure it is a dialect tsv file
+                if file.index("_") != file.rindex("_"):
+                    dialect_key = file[file.index("_") : file.rindex("_")]
+                    dialects = languages[iso639_code]["dialect"][dialect_key]
+                    if "|" in dialects:
+                        dialects = dialects.replace(" |", ",")
+                    wiki_name += f" ({dialects})"
+
+            row = [
+                iso639_code,
+                languages[iso639_code]["iso639_name"],
+                wiki_name,
+                str(languages[iso639_code]["casefold"]),
+                transcription_level,
+                str(num_of_entries),
+            ]
+
+            readme_tsv_list.append([file] + row)
+            readme_row_string = (
+                "| " + " | ".join([f"[TSV]({file})"] + row) + " |\n"
+            )
+            _readme_insert(wiki_name, readme_row_string)
+
+    # Write the TSV of the README
+    with open("readme_tsv.tsv", "w") as readme_tsv:
+        # Sort by wiktionary language name.
+        def sorting(ele):
+            return ele[3]
+        readme_tsv_list.sort(key=sorting)
+        for lang_row in readme_tsv_list:
```

Comment:
Strongly suggest you use `csv.writer` for this (you just need to tell it you're using a "\t" separator). Then call it's `writerows` method with `readme_tsv_list` as an argument and you're free of this loop too!